### ISSUE TITLE
 Product CSV Importer & Exporter - Image Alt, Title

### ISSUE THREAD

**Issue by yatgirl** [woocommerce/woocommerce#16669]
## Prerequisites

- [x ] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [ ] The issue still exists against the latest `master` branch of WooCommerce on Github
- [ ] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

1. Product CSV Importer & Exporter - when uploading images from an external url - no image title or alt text is added to media
2.
3.

## Expected/actual behavior

When I follow those steps, I see...
nothing other than the url in attachment details on the media screen.

I was expecting to see...
at least a title based on the image in the attachment details on the media screen. There is subsequently no alt text or title on product page.  
If this is expected behavior, may someone please advise the custom column code name for a custom import and export column.  I have 17000 products I just imported - I cant manually add alt text to each image....

## Isolating the problem

- [ ] This bug happens with only WooCommerce plugin active
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [x ] I can reproduce this bug consistently using the steps above

## WordPress Environment

<details>
```
Copy and paste the system status report from **WooCommerce > System Status** in WordPress admin here.
```
</details>

**Comment by claudiulodro** [comment]
We can make it so that alt text is generated based on the image name. It isn't possible to determine the alt text of an image from its URL, so we can't make the importer automatically figure out what the original alt text was.

Support for an alt text column may be worth exploring.

**Comment by yatgirl** [comment]
Yes - just putting the image name in the title and alt text would be much preferable to nothing.  I would be very grateful if you could please advise if I would be able to add this as a custom column to the importer - if so what is the code name of the column for image alt text or for the image title - I have been unable to find this information.

**Comment by claudiulodro** [comment]
@yatgirl the importer does not currently support a column for image alt text or title.

**Comment by yatgirl** [comment]
Yikes - ok thanks - I luckily have found a code snippet that will add the alt text to product images in the source code (doesnt add it to the media library) so that will have to suffice for now. 
https://stackoverflow.com/questions/27087772/how-can-i-change-meta-alt-and-title-in-catalog-thumbnail-product-thumbnail

**Comment by mikejolley** [comment]
Master/3.2 fixes the title issue already. Title will be set to the image filename. If your image has metadata, title and caption can be pulled in from the image itself.

In our code we have fallbacks for alt text if not set; caption, then post title. So alt text will be shown when it's output in our gallery. And if you use image metadata, caption can also be used.

tl;dr 3.2 has improved handling.

I don't feel we should overcomplicate the CSV import, as adding more fields like this would make the image field way more complicated than a list of URLs. If you have ideas on how to handle this data in a CSV without adding too much complication, please log this on ideas.woocommerce.com for feedback from other users before we take it on.

Thanks!
